thunder/examine/__init__.py

```diff
@@ -308,25 +332,51 @@ def make_trace_dot(trace: TraceCtx):
 
     roots, leaves = bsym_list_to_dag(trace.bound_symbols)
     leaves_id = {id(leaf) for leaf in leaves}
+    roots_id = {id(root) for root in roots}
+
+    def _get_color(node_id):
+        if node_id in roots_id:
+            return "green"
+        if node_id in leaves_id:
+            return "orange"
+        return "lightblue"
+
+    # All roots will be positioned at same level due to `rank`:`same`.
+    roots_g = graphviz.Digraph(graph_attr={"rank": "same"})
+    for root in roots:
+        roots_g.node(str(id(root)), root.bsym.python(indent=0, print_depth=1)[0], fillcolor=_get_color(str(id(root))))
+    dot.subgraph(roots_g)  # Add roots_g to main graph.
+
     stack = [*roots]
     visited = set()
+
+    # Breadth first
     while stack:
-        node: Node = stack.pop()
+        node: Node = stack.pop(0)
         node_id = id(node)
         visited.add(node_id)
-        dot.node(str(node_id), node.bsym.sym.name, fillcolor="orange" if node_id in leaves_id else "white")
-
+        color = _get_color(node_id)
+        dot.node(str(node_id), node.bsym.python(indent=0, print_depth=1)[0], fillcolor=color)
+
+        # Add node for args and connect args
+        for arg in node.bsym.flat_args:
+            if isinstance(arg, TensorProxy):
+                arg_id = arg.name
+                dot.node(arg_id, _repr_tensor_proxy(arg, show_metadata))
```

Comment:
My concern here is that with big graphs adding the node increases the time that it takes to render out the graph because it will try to layout more nodes that it is now. It's ok like this but I would want to try it first

Reply:
Creating a graph is a tool to aid understandability or debugging. It's okay if the graph generation is slow.

Reply:
I found a few options to improve the graph rendering times thought at the cost of more clumsy placement for nodes.

```python
dot = make_trace_dot(t)

# Update `nlimit` for less iterations of layout algorithm.
dot.graph_attr["nslimit"] = "5"
dot.graph_attr["nslimit1"] = "5"

# Or maybe use a different engine.
dot.render(format='png')
```

I don't know exactly how much they affect the times (and what are a good value) but we can potentially document it. Wdyt? Or maybe think about it in a follow-up.

Ref:
https://graphviz.org/docs/attrs/nslimit/
https://stackoverflow.com/questions/64130276/graphviz-is-too-slow

Reply:
Yeah I also experimented with those options at the start and they result in messier graphs. But I think using `nslimit` in worst case scenarios it's a good compromise! 

@IvanYashchuk we have the same idea here, tho I have some more context from playing around with graphviz and with the default renderer. I've seen it take 30 minutes to render out complex graphs so I just wanted to share the awareness on this and of course if it takes 2 minutes it's fine. So it's worth testing in my opinion

Reply:
I've quickly measured and it compares like this for the example of the bwd trace: 142,56s vs 58,62s of the previous version in main. Maybe we should have a "fast mode" that sets the nslimit and then the speedup is significant (2,79 seconds on same example!).